tests/integration/test_analysisd/test_signature_id/test_signature_id.py

```diff
@@ -0,0 +1,285 @@
+
+# Reference paths
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+RULES_SAMPLE_PATH = os.path.join(TEST_DATA_PATH, 'rules_samples')
+CONFIGURATIONS_PATH = os.path.join(TEST_DATA_PATH, 'configuration_template')
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
+
+# ----------------------------------------TEST_VALID_SIGNATURE_ID------------------------------------------
+# Configuration and cases data
+t1_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_signature_id_values.yaml')
+t1_cases_path = os.path.join(TEST_CASES_PATH, 'cases_valid_signature_id.yaml')
+
+# test_valid_signature_id configurations
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(t1_cases_path)
+t1_configurations = load_configuration_template(t1_configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
+
+# ----------------------------------------TEST_INVALID_SIGNATURE_ID------------------------------------------
+# Configuration and cases data
+t2_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_signature_id_values.yaml')
+t2_cases_path = os.path.join(TEST_CASES_PATH, 'cases_invalid_signature_id.yaml')
+
+# test_empty_signature_id configurations
+t2_configuration_parameters, t2_configuration_metadata, t2_case_ids = get_test_cases_data(t2_cases_path)
+t2_configurations = load_configuration_template(t2_configurations_path, t2_configuration_parameters,
+                                                t2_configuration_metadata)
+
+# ----------------------------------------TEST_NULL_SIGNATURE_ID------------------------------------------
+# Configuration and cases data
+t3_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_signature_id_values.yaml')
+t3_cases_path = os.path.join(TEST_CASES_PATH, 'cases_null_signature_id.yaml')
+
+# test_null_signature_id configurations
+t3_configuration_parameters, t3_configuration_metadata, t3_case_ids = get_test_cases_data(t3_cases_path)
+t3_configurations = load_configuration_template(t3_configurations_path, t3_configuration_parameters,
+                                                t3_configuration_metadata)
+
+
+# Tests
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_valid_signature_id(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                            prepare_custom_rules_file, restart_wazuh_function):
+    '''
+    description: Check that when a rule has an valid signature ID value assigned to the if_sid option, the rule is
+                 not ignored.
+
+    test_phases:
+        - Setup:
+            - Set wazuh configuration.
+            - Copy custom rules file into manager
+            - Clean logs files and restart wazuh to apply the configuration.
+        - Test:
```

Comment:
```suggestion
        - test:
```

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/3649/commits/80c472ee2c0de8b0c1f521d95117bedd0f8615a6